### PR TITLE
GSDX-Windows: Add "Large Framebuffer" option

### PR DESCRIPTION
**Pull request by ssakash**
**Summary of Changes**:

* Add ``Large Framebuffer`` option to windows GUI.

image

### CODE REVIEW

**Comment by gregory38** [comment]
Why not put it next to 8 bits texture ?

File: plugins/GSdx/GSSetting.cpp [hunk 1]
```diff
@@ -136,11 +136,9 @@ const char* dialog_message(int ID, bool* updateText) {
 		case IDC_FAST_TC_INV:
 			return "By default, the texture cache handles partial invalidations. Unfortunately it is very costly to compute CPU wise."
 				"\n\nThis hack replaces the partial invalidation with a complete deletion of the texture to reduce the CPU load.\n\nIt helps snowblind engine game.";
-#ifdef __linux__
 		case IDC_LARGE_FB:
-			return "Allocate a large framebuffer to be compliant with GS memory (help FMV flickering).\n"
-				"It increases GPU/memory requirements.";
-#endif
+			return "Allocate a large framebuffer to be compliant with GS memory (Prevents FMV flickering).\n"
+				"It increases VRAM requirements.";
```
Comment (gregory38) [inline plugins/GSdx/GSSetting.cpp]: It also increases GPU shader requirements (all games).

Comment (ssakash) [inline plugins/GSdx/GSSetting.cpp]: changed it back to ``GPU/memory``

**Comment by ssakash** [comment]
> Why not put it next to 8 bits texture ?

Initially I had thought it'd look a bit congested over there, but looks better than expected.

image2

**Comment by FlatOutPS2** [comment]
Is the large frame buffer only used for FMV's or all the time?

**Comment by ssakash** [comment]
> Is the large frame buffer only used for FMV's or all the time?

Large Frame buffer is used all the time.

**Comment by FlatOutPS2** [comment]
If used for fixing flickering FMV's, wouldn't a game fix be more sensible. It wouldn't kill performance during the rest of the game.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 2]
```diff
@@ -155,6 +155,7 @@ void GSSettingsDlg::OnInit()
 	ComboBoxInit(IDC_CRC_LEVEL, theApp.m_gs_crc_level, theApp.GetConfig("crc_hack_level", 3));
 
 	CheckDlgButton(m_hWnd, IDC_PALTEX, theApp.GetConfig("paltex", 0));
+	CheckDlgButton(m_hWnd, IDC_LARGE_FB, theApp.GetConfig("large_framebuffer", 0));
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.cpp]: Default should be on, not off?

**Comment by gregory38** [comment]
Option is a temporary solution (don't quote me on that next year) for people that use git build.
I'm confident we can have both speed and a large framebuffer.

**Comment by refractionpcsx2** [comment]
I don't think we really need a GUI option for a temporary option like this. I know we did with HW Depth, but that could have potentially stayed as a permanent option (of course, it didn't)

Doing ini entries should really be sufficient for this and use it for testing purposes to see how much it actually helps.

File: plugins/GSdx/GSSettingsDlg.cpp (hunk 2 above)
Comment (ssakash) [inline plugins/GSdx/GSSettingsDlg.cpp]: Personally I prefer having it disabled by default, but I see that gregory has it enabled on linux GUI. so went with that.

**Comment by gregory38** [comment]
It is faster to add the option than to explain how to edit the ini file with the good option name.

**Pull request merged**